packages/enzyme-adapter-utils/src/createMountWrapper.jsx

```diff
@@ -79,26 +87,53 @@ export default function createMountWrapper(node, options = {}) {
       return inst;
     }
 
+    setRootFinderInstance(instance) {
+      this.rootFinderInstance = instance;
+    }
+
+    getComponentParentInstance() {
+      // If a wrappingComponent was provided, the root node's parent will be the
+      // RootFinder component (see render()).
+      if (WrappingComponent) {
+        if (!this.rootFinderInstance) {
+          throw new Error('`wrappingComponent` must render its children!');
+        }
+
+        return this.rootFinderInstance;
+      }
+
+      // Otherwise, this component is the root node's parent!
+      return this;
+    }
+
     setChildContext(context) {
       return new Promise(resolve => this.setState({ context }, resolve));
     }
 
     render() {
       const { Component } = this.props;
-      const { mount, props } = this.state;
+      const { mount, props, wrappingComponentProps } = this.state;
       if (!mount) return null;
-      return (
-        <Component {...props} />
-      );
+      const component = <Component {...props} />;
+      if (WrappingComponent) {
+        return (
+          <WrappingComponent {...wrappingComponentProps}>
+            <RootFinder ref={this.setRootFinderInstance}>{component}</RootFinder>
```

Comment:
We need a reliable way to get the `node` that was passed as the first argument to `mount()` and this seemed like the simplest way. Is it acceptable? Are there going to be performance implications of using a `ref`?

Reply:
we should avoid using `ref`, yes.